src/calculateRentalCost.js

```diff
@@ -3,8 +3,29 @@
  *
  * @return {number}
  */
+
+const DAY_COST_VALUE = 40;
+const THREE_DAYS_OFF_VALUE = 20;
+const SEVEN_DAYS_OFF_VALUE = 50;
+
 function calculateRentalCost(days) {
-  // write code here
+  const isInvalidData = typeof days !== 'number';
+
+  if (isInvalidData) {
+    return 0;
+  }
+
+  const totalCost = DAY_COST_VALUE * days;
+
+  if (days >= 7) {
+    return totalCost - SEVEN_DAYS_OFF_VALUE;
```

Comment:
The check for the number of days should use a constant that explains the threshold for the discount, to improve code readability and maintainability.